### ISSUE TITLE
put w/signal warning with shmem_wait_until

### ISSUE THREAD

**Issue by davidozog** [Sandia-OpenSHMEM/SOS#814]
I've been seeing these compiler warnings lately when building the `shmemx` tests:
```
../../../test/shmemx/put_signal_nbi.c: In function 'main':
../../../test/shmemx/put_signal_nbi.c:70:5: warning: pointer targets in passing argument 1 of 'shmem_wait_until' differ in signedness [-Wpointer-sign]
     shmem_wait_until(&sig_addr, SHMEM_CMP_EQ, 1);
     ^
In file included from ../../../test/shmemx/put_signal_nbi.c:34:0:
../../mpp/shmem.h:1602:32: note: expected 'long int *' but argument is of type 'uint64_t *'
 SHMEM_FUNCTION_ATTRIBUTES void shmem_wait_until(long *ivar, int cmp, long value);
                                ^
  ../../../test/shmemx/put_signal.c: In function 'main':
../../../test/shmemx/put_signal.c:68:9: warning: pointer targets in passing argument 1 of 'shmem_wait_until' differ in signedness [-Wpointer-sign]
         shmem_wait_until(&sig_addr, SHMEM_CMP_EQ, 1);
         ^
In file included from ../../../test/shmemx/put_signal.c:35:0:
../../mpp/shmem.h:1602:32: note: expected 'long int *' but argument is of type 'uint64_t *'
 SHMEM_FUNCTION_ATTRIBUTES void shmem_wait_until(long *ivar, int cmp, long value);
                                ^
../../../test/shmemx/put_signal.c:70:9: warning: pointer targets in passing argument 1 of 'shmem_wait_until' differ in signedness [-Wpointer-sign]
         shmem_wait_until(&sig_addr, SHMEM_CMP_EQ, 1);
         ^
In file included from ../../../test/shmemx/put_signal.c:35:0:
../../mpp/shmem.h:1602:32: note: expected 'long int *' but argument is of type 'uint64_t *'
 SHMEM_FUNCTION_ATTRIBUTES void shmem_wait_until(long *ivar, int cmp, long value);
                                ^
```

It looks like this compiler (`gcc` in Travis's Trusty image) does not have C11 generic support and ends up using the legacy untyped API (with a `long` 1st argument) [here](https://github.com/Sandia-OpenSHMEM/SOS/blob/master/src/synchronization_c.c4#L147).

We might be better off using the `shmem_uint64_wait_until` in the `put_signal` and `put_signal_nbi` tests.

**Comment by wrrobin** [comment]
@davidozog Thanks for posting the issue. Will fix this in the optimized implementation.

**Comment by jdinan** [comment]
You can use the generic routine as long as you wrap it as follows:
```
#if defined(__STDC_VERSION__) && __STDC_VERSION__ >= 201112L
     shmem_wait_until(&sig_addr, SHMEM_CMP_EQ, 1);
#else
     shmem_uint64_wait_until(&sig_addr, SHMEM_CMP_EQ, 1);
#endif
```

**Comment by wrrobin** [comment]
Closed by #815
